WordPress/Classes/ViewRelated/System/WPSplitViewController.swift

```diff
@@ -0,0 +1,163 @@
+     *  navigation controller, so we'll pop off everything but the root view
+     *  controller and wrap it in a navigation controller if necessary.
+     */
+    func splitViewController(splitViewController: UISplitViewController, separateSecondaryViewControllerFromPrimaryViewController primaryViewController: UIViewController) -> UIViewController? {
+        guard let primaryNavigationController = primaryViewController as? UINavigationController else {
+            assertionFailure("Split view's primary view controller should be a navigation controller!")
+            return nil
+        }
+
+        // Grab all but the root view controller from the primary navigation stack,
+        // and pop it back.
+        let viewControllers = Array(primaryNavigationController.viewControllers.dropFirst())
+        primaryNavigationController.popToRootViewControllerAnimated(false)
+
+        // If we have no detail view controllers, try and fetch the primary view controller's
+        // initial detail view controller.
+        if viewControllers.count == 0 {
+            if let primaryViewController = primaryNavigationController.viewControllers.first,
+                let initialDetailViewController = initialDetailViewControllerForPrimaryViewController(primaryViewController) {
+                return initialDetailViewController
+            }
+        }
+
+        if let firstViewController = viewControllers.first as? UINavigationController {
+            // If it's already a navigation controller, just return it
+            return firstViewController
+        } else {
+            let navigationController = UINavigationController()
+            navigationController.viewControllers = viewControllers
+
+            return navigationController
+        }
+    }
+
+    func splitViewController(splitViewController: UISplitViewController, collapseSecondaryViewController secondaryViewController: UIViewController, ontoPrimaryViewController primaryViewController: UIViewController) -> Bool {
+        // If the user hasn't modified the navigation stack, then show the root view controller initially.
+        // In a horizontally compact size class this means we can collapse to show the root
+        // view controller, instead of having the detail view controller pushed onto the stack.
+        if let navigationController = primaryViewController as? UINavigationController where !primaryNavigationControllerStackHasBeenModified {
+            navigationController.popToRootViewControllerAnimated(false)
+            return true
+        }
+
+        return false
+    }
+}
+
+extension WPSplitViewController: UINavigationControllerDelegate {
+    func navigationController(navigationController: UINavigationController, willShowViewController viewController: UIViewController, animated: Bool) {
+        primaryNavigationControllerStackHasBeenModified = true
+    }
+}
+
+extension UIViewController {
+    /// Convenience method
+    var splitViewControllerIsCollapsed: Bool {
```

Comment:
As similarly noted above, maybe we could change this to read a bit more clearly as to the states it can reveal; perhaps multiple states as something like (roughly): `SplitViewFull`, `SplitViewCollapsed`, `SplitViewNone`.